src/components/AudioPlayer/RepeatButton.tsx

```diff
@@ -11,23 +12,28 @@ import { RepetitionMode } from './RepeatAudioModal/SelectRepetitionMode';
 import Badge from 'src/components/dls/Badge/Badge';
 import Button, { ButtonShape, ButtonVariant } from 'src/components/dls/Button/Button';
 import Wrapper from 'src/components/Wrapper/Wrapper';
-import {
-  selectIsInRepeatMode,
-  selectRemainingRangeRepeatCount,
-  selectAudioData,
-} from 'src/redux/slices/AudioPlayer/state';
 import { logButtonClick } from 'src/utils/eventLogger';
 import { toLocalizedNumber } from 'src/utils/locale';
+import { AudioPlayerMachineContext } from 'src/xstate/AudioPlayerMachineContext';
 
-const RepeatAudioButton = () => {
+const RemainingRangeCount = ({ rangeActor }) => {
   const { lang } = useTranslation('common');
+  const remainingCount = useSelector(rangeActor, (state) => {
+    const { totalRangeCycle, currentRangeCycle } = (state as any).context;
+    return totalRangeCycle - currentRangeCycle + 1; // +1 to include the current cycle
+  });
+  const localizedRemainingCount = toLocalizedNumber(remainingCount, lang);
+
+  return <span>{localizedRemainingCount}</span>;
+};
+
+const RepeatAudioButton = () => {
+  const audioService = useContext(AudioPlayerMachineContext);
+
   const [isModalOpen, setIsModalOpen] = useState(false);
-  const audioData = useSelector(selectAudioData, shallowEqual);
-  const isInRepeatMode = useSelector(selectIsInRepeatMode);
-  const remainingRangeRepeatCount = toLocalizedNumber(
-    useSelector(selectRemainingRangeRepeatCount),
-    lang,
-  );
+  const isInRepeatMode = useSelector(audioService, (state) => !!state.context.repeatActor);
+  const currentSurah = useSelector(audioService, (state) => state.context.surah);
+  const repeatActor = useSelector(audioService, (state) => state.context.repeatActor);
```

Comment:
This can be simplified as following:

```
  const currentSurah = useSelector(audioService, (state) => state.context.surah);
  const repeatActor = useSelector(audioService, (state) => state.context.repeatActor);
  const isInRepeatMode = !!repeatActor;
```